cpp/src/rolling/rolling_detail.hpp

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef ROLLING_DETAIL_HPP
+#define ROLLING_DETAIL_HPP
+
+#include <cudf/detail/utilities/device_operators.cuh>
+#include <cudf/utilities/traits.hpp>
+
+namespace cudf
+{
+
+// helper functions - used in the rolling window implementation and tests
+namespace detail
+{
+  // return true if ColumnType is arithmetic type or
+  // AggOp is min_op/max_op/count_op for wrapper (non-arithmetic) types
+  template <typename ColumnType, class AggOp, bool is_mean>
```

Comment:
The `is_mean` param seems redundant since `AggOp` is already available as compile time info.

Reply:
AggOp comes from device_operatores -- there's no mean operator. The division by count is computed on output (inherited from legacy implementation). So you can't tell that this is a mean from just the operator.